packages/report-server/src/reportBuilder/transform/aliases/summaryAliases.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Row } from '../../types';
+
+/**
+ * Inserts a new column with a value that is the calculation of the percentage of 1s out of 1s and 0s in a row.
+ * { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N'  }
+ *  => { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N', rowSummary: 0.75 }
+ */
+
+const detectColumnsToSummarise = (rows: Row[]): string[] => {
+  console.log(rows)
+  const anyRelevantColumns = rows.reduce((previousValue, currentValue) => {
+    const columnsWithYorN = Object.entries(currentValue).filter(([,value]) => value === 'Y' || value === 'N').map(([column,]) => column);
+    return previousValue.concat(columnsWithYorN);
+  },[] as string[])
+  const columnsToSummarise = anyRelevantColumns.reduce((previousValue: string[], currentValue: string): string[] => {
+    if (previousValue.indexOf(currentValue) > -1) {
```

Comment:
```suggestion
    if (previousValue.includes(currentValue)) {
```
`Array.includes()` is probably what you want to use here (achieves the same goal but reads a little nicer). 

However, overall it looks like you're just de-duplicating the array using this reduce? Probably my favourite way to do that these days is using a `Set`, eg. `const columnsToSummarise = [... new Set(anyRelevantColumns)];`

Better still, you can use a `Set` in the first reduce like so:
```
  const setOfColumnsToSummarise = rows.reduce((columnsToSummarise, row) => {
    const columnsWithYorN = Object.entries(currentValue).filter(([,value]) => value === 'Y' || value === 'N').map(([column,]) => column);
    columnsWithYorN.forEach(column => columnsToSummarise.add(column));
    return columnsToSummarise;
  }, new Set<string>());
  return [...setOfColumnsToSummarise];